packages/ra-core/src/types.ts

```diff
@@ -286,6 +286,8 @@ export type DataProviderProxy = {
 };
 
 export type MutationMode = 'pessimistic' | 'optimistic' | 'undoable';
+export type OnSuccess = (response?: any) => void;
+export type OnFailure = (error?: any) => void;
```

Comment:
We should probably avoid `any` in favor of `unknow`. See #5834

Reply:
I'm not sure I understand the benefits and shortcomings of `any` vs `unknown`. What I've read at https://stackoverflow.com/questions/51439843/unknown-vs-any tells me that there may be drawbacks for end users (like the obligation to perform type checking on an unknown value before using it).

Let's keep the two discussions separate.